### ISSUE TITLE
`stack` integration for `doctest`

### ISSUE THREAD

**Issue by Gabriel439** [sol/doctest#134]
I've been using `doctest` with `stack` and I notice that `doctest` does not pick the version of a package constrained by `stack`, even when run using `stack exec doctest`.

I wanted to ask what would be involved in setting up `doctest` to use the package versions that `stack` selects.  I'm not sure how `doctest` computes package versions to use when running the tests.

**Comment by sol** [comment]
@Gabriel439 Do you have a minimal example that illustrates your issue?

**Comment by Gabriel439** [comment]
I can no longer reproduce this, so I will close the issue, but I will report what I think my mistake was in case somebody else makes the same mistake I did.

The problem I observed went roughly like this:

In an empty directory, create the following two files:

```bash
$ cat stack.yaml
resolver: lts-5.11
packages: []
extra-deps:
- uniform-pair-0.1.10
- ShowF-0.1.1
```

```bash
$ cat Example.hs 
module Example where

import Data.UniformPair

swapP :: ()
swapP = ()

example :: ()
example = swapP
```

Then run:

```bash
$ stack build doctest uniform-pair
...
$ stack exec doctest Example.hs
Examples: 0  Tried: 0  Errors: 0  Failures: 0
```

That should work.  Now upgrade the `uniform-pair` in the `stack.yaml` to version `0.1.11`, which exports a newly added `swapP` function:

```bash
$ cat stack.yaml
resolver: lts-5.11
packages: []
extra-deps:
- uniform-pair-0.1.11
- ShowF-0.1.1
```

... now install that version and run the `doctest`, which fails as expected:

```bash
$ stack build uniform-pair
...
$ stack exec doctest Example.hs

Example.hs:9:11:
    Ambiguous occurrence ‘swapP’
    It could refer to either ‘Example.swapP’, defined at Example.hs:6:1
                          or ‘Data.UniformPair.swapP’,
                             imported from ‘Data.UniformPair’ at Example.hs:3:1-23
```

So far so good!

Now the problem I observed before was that `doctest` would be stuck on `uniform-pair-0.1.11` even if you downgraded the `stack.yaml` back to `uniform-pair-0.1.10` and run `stack build` (assuming that you had a real source project to build, unlike this minimal example).  However, in the course of creating the minimal reproducing example I noticed that the downgrade worked if I explicitly built `uniform-pair` like this:

```bash
$ stack build uniform-pair
uniform-pair-0.1.11: unregistering
uniform-pair-0.1.10: configure
uniform-pair-0.1.10: build
uniform-pair-0.1.10: copy/register
```

Then the downgrade worked just fine.

So I believe that the real problem that I was observing was that `stack` was not correctly unregistering `uniform-pair-0.1.11`in favor of `uniform-pair-0.1.10` when running `stack build` on my larger source project.

So, long story short, this was not really an issue with `doctest` but possibly an issue with `stack` (which I need to look into further).  I apologize for the incorrect bug report!
